File: web/src/utils/FilenameUtils.ts
export type Filename = {
  name: string;
  sequence: number;
  extension: string;
};

export function parseFilename(filename: string): Filename | null {
  const re = /^(.+?)[._-]([0-9]+)\.([^.]+)$/;
  const match = re.exec(filename);
  if (!match) {
    return null;
  }
  return {
    name: match[1],
    sequence: parseInt(match[2]),
    extension: match[3],
  };
}

File: server/src/main/kotlin/no/vargstudios/bifrost/server/util/FilenameUtils.kt
package no.vargstudios.bifrost.server.util

data class Filename(
    val name: String,
    val sequence: Int,
    val extension: String
)

fun parseFilename(filename: String): Filename? {
    val re = Regex("^(.+?)[._-]([0-9]+)\\.([^.]+)\$")
    val match = re.matchEntire(filename) ?: return null
    return Filename(
        match.groupValues[1],
        match.groupValues[2].toInt(),
        match.groupValues[3]
    )
}
